secret_handlers_test.go

```diff
@@ -489,3 +489,149 @@ func TestSecretListBadRequestInvalidFilter(t *testing.T) {
 
 	templates.BadRequestTest(t, rec)
 }
+
+func TestSecretGet(t *testing.T) {
+	testutils.SkipIfNotRunningIntegrationTests(t)
+	tenantIDForSecret := int64(1)
+	testUserId := "testUser"
+
+	userDao := dao.GetUserDao(&tenantIDForSecret)
+	user, err := userDao.FindOrCreate(testUserId)
+	if err != nil {
+		t.Error(err)
+	}
+
+	var userID *int64
+	for _, userScoped := range []bool{false, true} {
+		if userScoped {
+			userID = &user.Id
+		}
+
+		secret, err := dao.CreateSecretByName("Secret 1", &tenantIDForSecret, userID)
+		if err != nil {
+			t.Error(err)
+		}
+
+		secretID, err := util.InterfaceToString(secret.DbID)
+		if err != nil {
+			t.Error(err)
+		}
+
+		c, rec := request.CreateTestContext(
+			http.MethodGet,
+			"/api/sources/v3.1/secrets/"+secretID,
+			nil,
+			map[string]interface{}{
+				"tenantID": tenantIDForSecret,
+			},
+		)
+
+		c.SetParamNames("id")
+		c.SetParamValues(secretID)
+
+		tenancy := middleware.Tenancy(SecretGet)
+		userCatcher := middleware.UserCatcher(tenancy)
```

Comment:
I think the middleware.Tenancy() is not needed here, so this will work too
`userCatcher := middleware.UserCatcher(SecretGet)`